galsim/chromatic.py

```diff
@@ -131,6 +136,216 @@ def __init__(self, obj):
+
+        # Find the suggested image size for each object given the choice of scale, and use the
+        # maximum just to be safe.
+        possible_im_sizes = [ obj.SBProfile.getGoodImageSize(dx, 1.0) for obj in objs ]
+        n_im = max(possible_im_sizes)
+
+        # Find the stepK and maxK values for each object.  These will be used later on, so that we
+        # can force these values when instantiating InterpolatedImages before drawing.
+        self.stepK_vals = [ obj.stepK() for obj in objs ]
+        self.maxK_vals = [ obj.maxK() for obj in objs ]
+
+        # Finally, now that we have an image scale and size, draw all the images.  Note that
+        # `no_pixel` is used (we want the object on its own, without a pixel response).
+        self.ims = [ obj.drawImage(scale=dx, nx=n_im, ny=n_im, method='no_pixel') for obj in objs ]
+
+    def noInterpolation(self):
+        """
+        A routine to force an object for which interpolation was previously set up go back to the
+        exact calculation.
+        """
+        if hasattr(self, 'waves'):
+            # Check whether some chromatic transformation was done when setting up the
+            # interpolation, and restore the associated _A and _fluxFactor in combination with any
+            # other transformations that have been done in the meantime.
+            if hasattr(self, '_save_A'):
+                # Need to be careful about order.  What's in self._A came after what's in
+                # self._save_A.
+
+                if all([self._nullTransformation(w) for w in self.waves]):
+                    self._A = self._save_A
+                else:
+                    tmp_A = self._A
+                    self._A = lambda w: tmp_A(w) * self._save_A
+
+                if all([self._nullFluxTransformation(w) for w in self.waves]):
+                    self._fluxFactor = self._save_fluxFactor
+                else:
+                    tmp_fluxFactor = self._fluxFactor
+                    self._fluxFactor = lambda w: tmp_fluxFactor(w) * self._save_fluxFactor(w)
+
+                # Then delete the old _save_A.
+                del self._save_A
+                del self._save_fluxFactor
+
+            # Get rid of the stored attributes related to interpolation.
+            del self.waves
+            del self.stepK_vals
+            del self.maxK_vals
+            del self.ims
+
+    def hasInterpolation(self):
+        """
+        A routine to check whether an object has interpolation set up or not.
+
+        @returns    True or False
+        """
+        return hasattr(self, 'waves')
+
+    def _evaluateAtWavelength(self, wave):
+        """
+        Evaluate this ChromaticObject at a particular wavelength, either using interpolation or via
+        direct calculation, depending on whether or not interpolation has been set up.
+
+        @param wave     Wavelength in nanometers.
+
+        @returns the monochromatic object at the given wavelength, as a GSObject.
```

Comment:
It's not really wrong to have the doc string here, but it's kind of unnecessary.  Python won't display doc strings for private methods (leading underscore) when you say `help(ChromaticObject)` for instance.  It's generally sufficient to just use an in-code comment (hash comment) for private methods and functions.  And in that case, you don't need to include all the @param stuff when you are just documenting things for other developers.